api.py

```diff
@@ -59,6 +52,23 @@ def filter_data(self, dataframe):
         """
         return dataframe
 
+    def _convert_dates(self, df):
+        """
+        Ensure specified columns are typed as dates when inserted into the database.
+        """
+        if self.date_columns:
+            date_types = {col: "datetime64[ns]" for col in self.date_columns}
+            df = df.astype(date_types)
+        return df
+
+    def _rename_columns(self, df):
```

Comment:
The initial column names are also under our control, which makes this feel a little funny to me. Another way we could do this is in `preprocess_records`, where we already do some renaming / processing (e.g. converting the dueDate.year, etc. columns into one). What do you think?